### PR TITLE
Add translations No. 3

### PR DESCRIPTION
**Pull request by Echolon**
Hello @anurodhp & @tmolitor-stud-tu,

Third and last one - DONE! :tada: :tada: :tada: :tada: :beers:  :beers: :partying_face: 

Kind regards and looking forward!

rel. #144

### CODE REVIEW

File: Monal/Classes/MLNotificationSettingsViewController.m [hunk 1]
```diff
@@ -44,21 +44,21 @@ @implementation MLNotificationSettingsViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     
-    self.sectionsFooters =@[@"Apple push service should always be on. If it is off, your device can not talk to Apple's server.",
+    self.sectionsFooters =NSLocalizedString(@[@"Apple push service should always be on. If it is off, your device can not talk to Apple's server.",
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLNotificationSettingsViewController.m:47]: the translation has to be on the inner string, not the outer array ( `@[` starts an array while `@"` starts a string)

File: Monal/Classes/MLNotificationSettingsViewController.m [hunk 2]
```diff
@@ -44,21 +44,21 @@ @implementation MLNotificationSettingsViewController
...
                      @"If Monal can't show notifications, you will not see alerts when a message arrives. This happens if you tapped 'Decline' when Monal first asked permission.  Fix it by going to iOS Settings -> Monal -> Notifications and select 'Allow Notifications'. ",
                      @"If Monal push is off, your device could not talk to push.monal.im. This should also never be off. It requires Apple push service to work first. ",
                      @"",
-                            @"Rebuilding is useful if you are  expereicing problems. This will require an app restart to work."];
+                            @"Rebuilding is useful if you are  expereicing problems. This will require an app restart to work."],@ "");
     
-    self.sectionsHeaders =@[@"",
+    self.sectionsHeaders =NSLocalizedString(@[@"",
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLNotificationSettingsViewController.m:53]: same here: translate the strings, not the array

File: Monal/Classes/MLNotificationSettingsViewController.m [hunk 3]
```diff
@@ -44,21 +44,21 @@ @implementation MLNotificationSettingsViewController
...
                             @"",
                             @"",
                             @"Accounts",
-                            @"Advanced"];
+                            @"Advanced"],@ "");
     
-    self.apple=@[@"Apple Push Service"];
-    self.user=@[@"Can Show Notifications"];
-    self.monal=@[@"Monal Push Server"];
+    self.apple=NSLocalizedString(@[@"Apple Push Service"],@ "");
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLNotificationSettingsViewController.m:59]: here again...translate strings, not arrays

File: Monal/Classes/MLRegisterViewController.m [hunk 4]
```diff
@@ -139,7 +139,7 @@ -(IBAction)registerAccount:(id)sender {
                 
                 [[DataLayer sharedInstance] addAccountWithDictionary:dic andCompletion:^(BOOL result) {
                     if(result) {
-                        [[DataLayer sharedInstance] executeScalar:@"select max(account_id) from account" withCompletion:^(NSObject * accountid) {
+                        [[DataLayer sharedInstance] executeScalar:NSLocalizedString(@"select max(account_id) from account",@ "") withCompletion:^(NSObject * accountid) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLRegisterViewController.m:142]: this is a database command, don't translate anything in DataLayer.m

File: Monal/Classes/MLServerDetails.m [hunk 5]
```diff
@@ -32,32 +32,32 @@ -(void) viewWillAppear:(BOOL)animated
     
     if(self.xmppAccount.connectionProperties.supportsSM3)
     {
-        [self.serverCaps addObject:@{@"Title":@"XEP-0198: Stream Management", @"Description":@"Resume a stream when disconnected. Results in faster reconnect and saves battery life."}];
+        [self.serverCaps addObject:NSLocalizedString(@{@"Title":@"XEP-0198: Stream Management", @"Description":@"Resume a stream when disconnected. Results in faster reconnect and saves battery life."}];
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLServerDetails.m:35]: translate strings, not dictionaries (dictionaries start with `@{`, strings use `@"`)

File: Monal/Classes/MLServerDetails.m [hunk 6]
```diff
@@ -32,32 +32,32 @@ -(void) viewWillAppear:(BOOL)animated
...
     if(self.xmppAccount.connectionProperties.supportsHTTPUpload)
     {
-           [self.serverCaps addObject:@{@"Title":@"XEP-0363: HTTP File Upload", @"Description":@"Upload files to the server to share with others."}];
+           [self.serverCaps addObject:NSLocalizedString(@{@"Title":@"XEP-0363: HTTP File Upload", @"Description":@"Upload files to the server to share with others."}];
     }
     
     if(self.xmppAccount.connectionProperties.supportsClientState)
     {
-           [self.serverCaps addObject:@{@"Title":@"XEP-0352: Client State Indication", @"Description":@"Indicate when a particular device is active or inactive. Saves battery. "}];
+           [self.serverCaps addObject:NSLocalizedString(@{@"Title":@"XEP-0352: Client State Indication", @"Description":@"Indicate when a particular device is active or inactive. Saves battery."}];
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLServerDetails.m:60]: everything in between the last comment and this one tries to translate dictionaries, please fix

File: Monal/Classes/MLSettingsTableViewController.m [hunk 7]
```diff
@@ -38,14 +38,14 @@ - (IBAction)close:(id)sender
 - (void)viewDidLoad {
     [super viewDidLoad];
     
-    self.sections =@[@"App", @"Support", @"About"];
+    self.sections =NSLocalizedString(@[@"App", @"Support", @"About"],@ "");
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLSettingsTableViewController.m:41]: here again...translate strings, not arrays

File: Monal/Classes/MLSettingsTableViewController.m [hunk 8]
```diff
@@ -38,14 +38,14 @@ - (IBAction)close:(id)sender
...
-    self.appRows=@[@"Quick Setup", @"Accounts", @"Notifications", @"Backgrounds", @"Sounds", @"Display", @"Chat Logs"];  //@"Cloud Storage"
-    self.supportRows=@[@"Email Support", @"Submit A Bug"];
+    self.appRows=NSLocalizedString(@[@"Quick Setup", @"Accounts", @"Notifications", @"Backgrounds", @"Sounds", @"Display", @"Chat Logs"],@ "");  //@"Cloud Storage"
+    self.supportRows=@[@"Email Support", @"Submit A Bug"],@ "");
 #ifdef DEBUG
-    self.aboutRows=@[@"Rate Monal", @"Open Source", @"Privacy", @"Crash Logging",  @"About", @"Version",  @"Log" ];
+    self.aboutRows=NSLocalizedString(@[@"Rate Monal", @"Open Source", @"Privacy", @"Crash Logging",  @"About", @"Version",  @"Log" ],@ "");
 #else
-    self.aboutRows=@[@"Rate Monal", @"Open Source", @"Privacy", @"Crash Logging", @"About", @"Version"];
+    self.aboutRows=NSLocalizedString(@[@"Rate Monal", @"Open Source", @"Privacy", @"Crash Logging", @"About", @"Version"],@ "");
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLSettingsTableViewController.m:48]: here again...translate strings, not arrays

File: Monal/Classes/MLComposeViewController.m [hunk 9]
```diff
@@ -28,8 +28,8 @@ - (IBAction)send:(id)sender
 {
     if([[MLXMPPManager sharedInstance].connectedXMPP count]==0)
     {
-        UIAlertController *messageAlert =[UIAlertController alertControllerWithTitle:@"No connected accounts" message:@"Please make sure at least one account has connected before trying to message someone." preferredStyle:UIAlertControllerStyleAlert];
-        UIAlertAction *closeAction =[UIAlertAction actionWithTitle:@"Close" style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
+        UIAlertController *messageAlert =[UIAlertController alertControllerWithTitle:NSLocalizedString(@"No connected accounts",@"") message:NSLocalizedString(@"Please make sure at least one account has connected before trying to message someone.",@"") preferredStyle:UIAlertControllerStyleAlert];
+        UIAlertAction *closeAction =[UIAlertAction actionWithTitle:@"Close",@"") style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLComposeViewController.m:32]: you missed the `NSLocalizedString(` here

File: Monal/Classes/MLComposeViewController.m [hunk 10]
```diff
@@ -40,8 +40,8 @@ - (IBAction)send:(id)sender
         
         if(self.message.text.length==0)
         {
-            UIAlertController *messageAlert =[UIAlertController alertControllerWithTitle:@"Error" message:@"Message can't be empty" preferredStyle:UIAlertControllerStyleAlert];
-            UIAlertAction *closeAction =[UIAlertAction actionWithTitle:@"Close" style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
+            UIAlertController *messageAlert =[UIAlertController alertControllerWithTitle:NSLocalizedString(@"Error",@"") message:NSLocalizedString(NSLocalizedString(@"Message can't be empty",@"") preferredStyle:UIAlertControllerStyleAlert];
+            UIAlertAction *closeAction =[UIAlertAction actionWithTitle:@"Close",@"") style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLComposeViewController.m:44]: also missing here

**Comment by Echolon** [comment]
@tmolitor-stud-tu I think thats it :+1:

**Comment by Echolon** [comment]
There is no action from my side correct?

**Comment by anurodhp** [comment]
I haven’t looked yet, sorry

**Comment by Echolon** [comment]
No rush just asking in case I forgot something

**Comment by anurodhp** [comment]
i think it conflicts with the other PR .ive merged  you will have to rebase again

**Comment by Echolon** [comment]
@anurodhp should be fine now if everything is right

**Pull request merged**